disnake/guild.py

```diff
@@ -2600,8 +2602,13 @@ def fetch_members(
             members = await guild.fetch_members(limit=150).flatten()
             # members is now a list of Member...
         """
-        if not self._state._intents.members:
-            raise ClientException("Intents.members must be enabled to use this.")
+        # `hasattr` check to avoid issues with uninitialized state
+        if hasattr(self._state, "application_flags"):
+            flags = self._state.application_flags
+            if not (flags.gateway_guild_members_limited or flags.gateway_guild_members):
+                raise ClientException(
+                    "The `members` intent must be enabled in the Developer Portal to be able to use this method."
+                )
 
         return MemberIterator(self, limit=limit, after=after)
```

Comment:
What is the new error that can be run into here if the `if` is not triggered and intents are not defined? Will it be `HTTPException` and if so should that have handling to raise a `ClientException` instead? Otherwise possibly an amendment be made to the docstring such that the exception could also be due to intents

Reply:
Without member intents enabled in the dev portal, this endpoint throws a 403, which translates into `disnake.Forbidden`, i.e. a `HTTPException`.
The `if` check here really shouldn't be needed, I'm just being extra cautious. `application_flags` exists once the bot receives the `READY` event from the gateway, so this only concerns situations where you try to fetch members without connecting to the gateway first.